lib/reversetunnel/transport.go

```diff
@@ -119,6 +120,8 @@ func (t *TunnelAuthDialer) DialContext(ctx context.Context, _, _ string) (net.Co
 // payload.
 func parseDialReq(payload []byte) *sshutils.DialReq {
 	var req sshutils.DialReq
+	req.ClientSrcAddr = &net.TCPAddr{}
+	req.ClientDstAddr = &net.TCPAddr{}
```

Comment:
What's this (and the later check) for? If `json.Unmarshal` deals badly with `net.TCPAddr` then maybe we need a different type in `DialReq`.

Reply:
It comes from `ClientSrc\DstAddr` being an interface (`net.Addr`). But I think it created a bit too much awkward code and I'll rewrite it to be just simple `string`, should be much simpler and more compact.

Reply:
Rewrote